dll/win32/iphlpapi/ipstats_reactos.c

```diff
@@ -387,17 +386,18 @@ DWORD getNumRoutes(void)
             memset( &isnmp, 0, sizeof( isnmp ) );
             status = tdiGetMibForIpEntity( tcpFile, &entitySet[i], &isnmp );
             if( !NT_SUCCESS(status) ) {
-                tdiFreeThingSet( entitySet );
-                closeTcpFile( tcpFile );
-                return status;
+                numRoutes = 0;
+                goto NoMib;
```

Comment:
Put tdiGetEntityIDSet back as it was, set numRoutes = 0 and just use a break statement to exit the loop.

Reply:
Agreed. Done.